File: reactive.infra.form.engine/src/main/java/org/infra/reactive/form/engine/form/engine/model/dto/request/window/editors/DataProviderSearchRequestDTO.java
package org.infra.reactive.form.engine.form.engine.model.dto.request.window.editors;

import lombok.Data;
import org.infra.reactive.form.engine.form.engine.model.dto.request.common.PagingDTO;
import org.infra.reactive.form.engine.form.engine.model.dto.response.translate.CoreTranslateLanguageDTO;

import java.io.Serializable;
import java.util.Map;

@Data
public class DataProviderSearchRequestDTO {
    private Long id;
    private Long coreWindowTabId;
    private Long coreWindowTabFieldId;
    private Long coreProcessParamId;
    private String rawText;
    private Map<Long, Serializable> fieldValues;
    private PagingDTO pagingDTO;
    private CoreTranslateLanguageDTO coreTranslateLanguageDTO;
}

File: reactive.infra.web/src/main/resources/static/Resources/JS/ERPFrameWork/Communicate/Models/Request/Window/Editors/DataProviderSearchRequestDTO.js
import {BaseModel} from "../../../../../../UIFrameWork/Shared/Common/BaseModel.js";

export default class DataProviderSearchRequestDTO extends BaseModel {
    constructor() {
        super();
    }

    setCoreWindowTabId(coreWindowTabId) {
        this.coreWindowTabId = coreWindowTabId;
    }

    setCoreWindowTabFieldId(coreWindowTabFieldId) {
        this.coreWindowTabFieldId = coreWindowTabFieldId;
    }

    setCoreProcessParamId(coreProcessParamId) {
        this.coreProcessParamId = coreProcessParamId;
    }

    setRawText(rawText) {
        this.rawText = rawText;
    }

    setFieldValues(fieldValues) {
        this.fieldValues = fieldValues;
    }

    setPagingDTO(pagingDTO) {
        this.pagingDTO = pagingDTO;
    }
}